server/src-lib/Hasura/Events/Lib.hs

```diff
@@ -214,18 +219,20 @@ tryWebhook pool e _ = do
     Nothing -> return $ Left $ HOther "table or event-trigger not found"
     Just et -> do
       let webhook = etiWebhook et
+          headerConfs = fromMaybe [] $ etiHeaders et
           createdAt = eCreatedAt e
           eventId =  eId e
       eeCtx <- asks getter
-
+      mheaders <- liftIO $ mapM getHeader headerConfs
```

Comment:
Don't read these in every webhook call? We don't expect them to change once the server starts. So you might as well validate them and store the resolved header in the cache.